components/support/nimbus-fml/src/editing/error_kind.rs

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use std::collections::BTreeSet;
+
+use serde_json::{Map, Value};
+
+use crate::intermediate_representation::{PropDef, TypeRef};
+
+pub(crate) enum ErrorKind {
+    InvalidKey(TypeRef, /* found */ BTreeSet<String>),
+    InvalidPropKey(
+        /* valid */ BTreeSet<String>,
+        /* found */ BTreeSet<String>,
+    ),
```

Comment:
Can we use enums with struct members ?

```
    InvalidPropKey{
        valid: BTreeSet<String>,
        found: BTreeSet<String>,
    },
```

I'm not a super big fan of tuple-struct enum variants, especially when there are more than member *and* they share the same type.

Reply:
Yes, I had wondered if the tuple form was a bit too terse.